chain-tx-enclave-next/enclave-ra/ra-sp-client/src/client.rs

```diff
@@ -76,7 +77,17 @@ impl SpRaClient {
             .ok_or(SpRaClientError::NoResponse)?;
 
         match response {
-            Response::GetAttestationReport { attestation_report } => Ok(attestation_report),
+            Response::GetAttestationReport { attestation_report } => {
+                // validate nonce
```

Comment:
Unfortunate that we have to deserialize here, and then downstream, but it's because the nonce is not available to the caller in the other crates